source/MaterialXRender/ImageHandler.cpp

```diff
@@ -19,43 +19,148 @@ const string VADDRESS_MODE_SUFFIX(IMAGE_PROPERTY_SEPARATOR + "vaddressmode");
 const string FILTER_TYPE_SUFFIX(IMAGE_PROPERTY_SEPARATOR + "filtertype");
 const string DEFAULT_COLOR_SUFFIX(IMAGE_PROPERTY_SEPARATOR + "default");
 
-string ImageDesc::BASETYPE_UINT8 = "UINT8";
-string ImageDesc::BASETYPE_HALF = "HALF";
-string ImageDesc::BASETYPE_FLOAT = "FLOAT";
+const string ImageLoader::BMP_EXTENSION = "bmp";
+const string ImageLoader::EXR_EXTENSION = "exr";
+const string ImageLoader::GIF_EXTENSION = "gif";
+const string ImageLoader::HDR_EXTENSION = "hdr";
+const string ImageLoader::JPG_EXTENSION = "jpg";
+const string ImageLoader::JPEG_EXTENSION = "jpeg";
+const string ImageLoader::PIC_EXTENSION = "pic";
+const string ImageLoader::PNG_EXTENSION = "png";
+const string ImageLoader::PSD_EXTENSION = "psd";
+const string ImageLoader::TGA_EXTENSION = "tga";
+const string ImageLoader::TIF_EXTENSION = "tif";
+const string ImageLoader::TIFF_EXTENSION = "tiff";
+const string ImageLoader::TX_EXTENSION = "tx";
+const string ImageLoader::TXT_EXTENSION = "txt";
+const string ImageLoader::TXR_EXTENSION = "txr";
 
-string ImageDesc::IMAGETYPE_2D = "IMAGE2D";
+//
+// Image methods
+//
+
+Image::Image(unsigned int width, unsigned int height, unsigned int channelCount, BaseType baseType) :
+    _width(width),
+    _height(height),
+    _channelCount(channelCount),
+    _baseType(baseType),
+    _resourceBuffer(nullptr),
+    _resourceBufferDeallocator(nullptr),
+    _resourceId(0)
+{
+}
+
+Image::~Image()
+{
+    releaseResourceBuffer();
+}
+
+unsigned int Image::getBaseStride() const
+{
+    if (_baseType == BaseType::HALF)
+    {
+        return 2;
+    }
+    if (_baseType == BaseType::FLOAT)
+    {
+        return 4;
+    }
+    return 1;
+}
 
-void ImageDesc::freeResourceBuffer()
+unsigned int Image::getMaxMipCount() const
 {
-    if (resourceBuffer)
+    return (unsigned int) std::log2(std::max(_width, _height)) + 1;
+}
+
+ImagePtr Image::createConstantColor(unsigned int width, unsigned int height, const Color4& color)
+{
+    unsigned int channelCount = 4;
+    size_t bufferSize = width * height * channelCount;
+    if (!bufferSize)
     {
-        if (resourceBufferDeallocator)
+        return nullptr;
+    }
+
+    ImagePtr image = create(width, height, channelCount, Image::BaseType::FLOAT);
+    image->createResourceBuffer();
+    float* pixel = static_cast<float*>(image->getResourceBuffer());
+    for (size_t i = 0; i < image->getWidth(); i++)
+    {
+        for (size_t j = 0; j < image->getHeight(); j++)
         {
-            resourceBufferDeallocator(resourceBuffer);
+            for (unsigned int c = 0; c < image->getChannelCount(); c++)
+            {
+                *pixel++ = color[c];
+            }
+        }
+    }
+    return image;
+}
+
+Color4 Image::getTexelColor(unsigned int x, unsigned int y) const
```

Comment:
This doesn't seem to be used anywhere. I'm just curious about where this is intended to be used?